packages/extension_google_sign_in_as_googleapis_auth/example/lib/main.dart

```diff
@@ -38,37 +37,75 @@ class SignInDemo extends StatefulWidget {
 
 /// The state of the main widget.
 class SignInDemoState extends State<SignInDemo> {
+  late Future<void> _signInInitialized;
   GoogleSignInAccount? _currentUser;
+  GoogleSignInClientAuthorization? _authorization;
   String _contactText = '';
 
   @override
   void initState() {
     super.initState();
-    _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount? account) {
+
+    final GoogleSignIn signIn = GoogleSignIn.instance;
+    _signInInitialized = signIn.initialize(
+        // Add your client IDs here as necessary for your supported platforms.
+        );
+    signIn.authenticationEvents.listen((GoogleSignInAuthenticationEvent event) {
       setState(() {
-        _currentUser = account;
+        switch (event) {
+          case GoogleSignInAuthenticationEventSignIn():
+            _currentUser = event.user;
+          case GoogleSignInAuthenticationEventSignOut():
+            _currentUser = null;
+            _authorization = null;
+        }
       });
+
       if (_currentUser != null) {
-        _handleGetContact();
+        _checkAuthorization();
       }
+    }).onError((Object error) {
+      debugPrint(error.toString());
+    });
+
+    _signInInitialized.then((void value) {
+      signIn.attemptLightweightAuthentication();
     });
-    _googleSignIn.signInSilently();
   }
 
-  Future<void> _handleGetContact() async {
+  void _updateAuthorization(GoogleSignInClientAuthorization? authorization) {
     setState(() {
-      _contactText = 'Loading contact info...';
+      _authorization = authorization;
     });
 
-// #docregion CreateAPIClient
-    // Retrieve an [auth.AuthClient] from the current [GoogleSignIn] instance.
-    final auth.AuthClient? client = await _googleSignIn.authenticatedClient();
+    if (authorization != null) {
+      unawaited(_handleGetContact(authorization));
+    }
+  }
 
-    assert(client != null, 'Authenticated client missing!');
+  Future<void> _checkAuthorization() async {
+    _updateAuthorization(
+        await _currentUser?.authorizationClient.authorizationForScopes(scopes));
+  }
+
+  Future<void> _requestAuthorization() async {
+    _updateAuthorization(await _currentUser?.authorizationClient
+        .authorizeScopes(<String>[PeopleServiceApi.contactsReadonlyScope]));
+  }
+
+  Future<void> _handleGetContact(
+      GoogleSignInClientAuthorization authorization) async {
+    setState(() {
```

Comment:
There's a lot of asynchrony here and above that could be problematic. Do you think it's reasonable to add `if (mounted)` before some of these `setState()` calls?